### DISCUSSION TITLE
Auto Summary Functionality (And maybe Tagging)

### DISCUSSION THREAD

**Discussion by Keithsc** [open-webui/open-webui#5043] [Ideas]
I would like to propose a feature that could significantly enhance the usability of the "open-webui" project, especially during extended development sessions or conversations involving multiple topics. 
Problem Statement 

After chatting through speech-to-text (STT) functionality and discussing various topics, the resulting text can become lengthy and unwieldy. This is especially challenging when working on any coding task, where clear communication is essential. Often, these discussions culminate in conclusions or refined ideas, and users need a way to capture these insights without losing the context of their prior chats. 
Proposed Solution 

I propose the development of an "Auto Summary" feature that would be used at the end of a chat session, allowing users to distill their discussions into a concise summary and also tag the conversation for future reference: 

    Summary Button : Introduce a clear button labeled "Summarize" that, when pressed, sends the current chat text to the back-end model (or potentially a user-selectable model). 

    Detail Slider : Include a slider that allows users to select the level of detail for the summary, ranging from 1 (very brief) to 10 (detailed). 

    Tagging Option : After generating the summary, provide users with the option to automatically generate tags based on the content or allow them to manually input their tags. This could help in organizing discussions for future reference. 

    Response Handling : 
        Once the summary and tags are generated, display them in a user-friendly format.
        If users choose to accept the summary and tags, the original chat text would be archived, and the summary and tags would replace the original chat in the chat interface, allowing the important insights to be retained and easily accessible.
         
     

Benefits 

    Enhanced Clarity : Enables users to distill long discussions into manageable summaries, improving readability and focus.
    Increased Efficiency : Provides a quick way to reference key points without having to scroll through extensive chat histories.
    Retention of Key Insights : Users can capture important conclusions and ideas from their discussions without losing them, avoiding the pitfall of archiving that makes retrieval difficult.
    Improved Organization : The option to auto-generate or manually input tags allows users to easily categorize and find relevant chats in the future.
    Unique Feature : This auto-summary and tagging capability is a distinctive offering that is currently not available in other chat-based systems, making it a potential killer feature for "open-webui."
     

I believe this feature could greatly enhance the user experience by simplifying interactions, facilitating better organization of discussions, and aiding in efficient development processes. Thank you for considering this request.

**Comment by mdlmarkham** [comment]
I really like this idea - perhaps there could also be a way to incorporate the summary into the knowledge base of the model, or to add it to the documents section.

**Comment by BenceSzalai** [comment]
So what this really needs to make it happen? Just someone working on it and making a PR or are there any concerns or obstacles to clear first?

**Comment by Davixk** [comment]
this would work great with the Previous Chats recall feature #13595, as the WebUI could treat past chats as Memories and retrieve relevant ones

tnother perk of this would be, that in the future if we'll implement the Context Limiter #6692 , then this could be use to overcome context limitations by summarizing chats with inexpensive models, and continuing them with your currently selected model
